src/components/Dialog.tsx

```diff
@@ -51,18 +50,43 @@ export default class Dialog extends Component<DialogProps, DialogState> {
   getSourceIDAndPaths = async (
     props: DialogProps,
   ): Promise<Array<SourceProps>> => {
-    const sources = await this.getSources(props);
-    const enabledSources = sources.data.reduce(
-      (result: SourceProps[], source: any) => {
-        if (source.attributes.enabled) {
-          let id = source.id;
-          let name = source.attributes.name;
-          result.push({ id, name });
-        }
-        return result;
-      },
-      [],
-    );
+    let sources,
+      enabledSources: Array<SourceProps> = [];
+
+    try {
+      sources = await this.getSources(props);
+    } catch (error) {
+      console.error(error);
+      return enabledSources;
```

Comment:
_Maybe_, it's better to `throw` here?

Reply:
I want to make sure any issues like this to cause the application to fail gracefully, throwing here will cause the application to crash. Also if I can narrow the error type down to `APIError`, then I can emit more useful error for support purposes.